src/Compilers/CSharp/Portable/Binder/ExtensionTypeScopes.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// An enumerable collection of extension type scopes in search
+    /// order, from the given Binder, out through containing Binders.
+    /// </summary>
+    internal readonly struct ExtensionTypeScopes
```

Comment:
>ExtensionTypeScopes [](http://example.com/codeflow?start=29&length=19)

This looks like a complete clone of ```ExtensionMethodScopes```, with only difference that the ```ExtensionMethodScopes``` uses a wrapper around the ```Binder``` type, ```ExtensionMethodScope```. I think we should simply reuse the ```ExtensionMethodScopes```. Perhaps renaming it to ```ExtensionScopes```, ```ExtensionMethodScope``` to ```ExtensionScope```, and ```Binder.SupportsExtensionMethods``` to ```SupportsExtensions```.